ddtrace/_trace/pin.py

```diff
@@ -138,12 +119,11 @@ def override(
     @classmethod
     def _override(
         cls,
-        obj,  # type: Any
-        service=None,  # type: Optional[str]
-        tags=None,  # type: Optional[Dict[str, str]]
-        tracer=None,
-    ):
-        # type: (...) -> None
+        obj: Any,
+        service: Optional[str] = None,
+        tags: Optional[Dict[str, str]] = None,
+        tracer: Optional[Any] = None,
```

Comment:
This change resolves a circular loop that will allow us to use the `Tracer` type here: https://github.com/DataDog/dd-trace-py/pull/13109.

Reply:
Should we consider having a `Tracer` abstract type that we can use for typing?

Reply:
Yup that's a great idea. We should avoid exposing implementation details where possible. I think we should have an abstract type for the `Tracer`, `Span`, and `Context` objects. I'll add this to my backlog.